### PR TITLE
Add Submariner known issues

### PR DESCRIPTION
**Pull request by dfarrell07**
Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

### CODE REVIEW

**Review by aswinsuryan — APPROVED** [review]

File: release_notes/known_issues.adoc [hunk 1]
```diff
@@ -935,12 +935,22 @@ To address the previously mentioned limitations, when a `restore.cluster.open-cl
...
 [#submariner-no-rhel]
 === Submariner currently does not support Red Hat Enterprise Linux nodes as worker nodes
 
 When deploying Submariner on a cluster that includes Red Hat Enterprise Linux worker nodes, application workloads fail to communicate with remote clusters. Submariner only supports {ocp-short} clusters that use Red Hat Enterprise Linux CoreOS nodes.
 
-[#mtu-issue]
+[#submariner-not-all-infra]
+=== Submariner currently does not support all infrastructure providers ACM can manage
+
+Not all infrastructure providers ACM can manage are supported with Submariner. Please refer to the support matrix page for more details.
```
Comment (sridhargaddam) [inline release_notes/known_issues.adoc]:
```suggestion
Not all infrastructure providers ACM can manage are supported with Submariner. Please refer to the [support matrix](link to support matrix page) page for more details.
```

Might be good to link the support matrix page.

File: release_notes/known_issues.adoc [hunk 2]
```diff
@@ -955,3 +965,18 @@ Replace `remote-ep-cidr` with the the global CIDR of the remote endpoint.
...
+
+[#submariner-openstack-ui]
+=== Submariner currently does not support preparing OpenStack infrastructure from the ACM UI
+
+Automatic cloud preparation for OpenStack clusters is currently not supported from the ACM UI. However, it works fine if the ACM APIs are used.
+
+[#submariner-headless-services-globalnet]
+=== Submariner currently does not support Headless Services with Globalnet
+
+When using Headless Services with Globalnet, DNS resolution for local pods doesn’t work correctly.
```
Comment (sridhargaddam) [inline release_notes/known_issues.adoc]:
```suggestion
When using Headless Services with Globalnet, DNS resolution for local pods doesn’t work correctly when using clusterset.local domain name.
```

File: release_notes/known_issues.adoc (hunk 1 above)
Comment (nyechiel) [reply release_notes/known_issues.adoc]: Not sure what we do in other instances across the docs, but note that the support matrix page is going to change with each ACM version. So if we link the 2.5 matrix here, we also need to ensure we update the URL with each new ACM version. Unless we have a way to automate this, maybe it's better not to link for now?

**Comment by nyechiel** [comment]
The content seems accurate to me, but we need a tech writer to review and edit for style and consistency, etc.

File: release_notes/known_issues.adoc (hunk 1 above)
Comment (sridhargaddam) [reply release_notes/known_issues.adoc]: Okay in that case, agree we can leave it in the current state.

**Review by cadawson — CHANGES_REQUESTED** [review]
Thanks. I made some minor edits that should not have changed the technical meaning, but made it more consistent with our docs. Please take a look and make sure that it looks ok. 
Thanks!

**Comment by cadawson** [comment]
I went ahead and added the link to the support matrix for convenience. It looks like we mention in the Submariner section of the docs which environments are supported for using Submariner at: https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.5/html/add-ons/add-ons-overview#submariner. Do we need this additional known issue to mention it? Thanks. @dfarrell07

**Comment by nyechiel** [comment]
> I went ahead and added the link to the support matrix for convenience. It looks like we mention in the Submariner section of the docs which environments are supported for using Submariner at: https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.5/html/add-ons/add-ons-overview#submariner. Do we need this additional known issue to mention it? Thanks. @dfarrell07

IMO it's better to have this mentioned here as well.

File: release_notes/known_issues.adoc (hunk 1 above)
Comment (cadawson) [reply release_notes/known_issues.adoc]: Added the link, for convenience. Question added to the comments.

File: release_notes/known_issues.adoc (hunk 2 above)
Comment (cadawson) [reply release_notes/known_issues.adoc]: Made this change manually in the doc, so I covered the edits, as well.

**Comment by dfarrell07** [comment]
These edits look great to me, thanks @cadawson! Tangentially, I think this PR is a perfect example of how the process should work: dev write content they are expert in, docs writer makes it RH-style-compliant.

**Comment by cadawson** [comment]
Thanks all! I think this worked well, too. Time to start looking at the automation!

**Review by cadawson — APPROVED** [review]
Looks good! Thanks!

**Comment by cadawson** [comment]
Issue #23508 created for this.

**Pull request merged**